src/App.tsx

```diff
@@ -14,11 +16,208 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  showGoods: boolean,
+  goods: string[],
+  isReversed: boolean,
+  isSorted: boolean,
+  sortBy: string,
+  minLength: number,
+}
+
+class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    showGoods: false,
+    goods: goodsFromServer,
+    isReversed: false,
+    isSorted: false,
+    sortBy: '',
+    minLength: 1,
+  };
+
+  showGoodsList = () => {
+    this.setState({ showGoods: true });
+  };
+
+  sortAlphabetically = () => {
+    this.setState({ isSorted: true, sortBy: 'alphabet' });
+  };
+
+  sortByLength = () => {
+    this.setState({ isSorted: true, sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      isSorted: false,
+      sortBy: '',
+      minLength: 1,
+    });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  setMinLength = (value: number) => {
+    this.setState({ minLength: value });
+  };
```

Comment:
use correct naming for event handlers: 
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364